### PR TITLE
Add instructions for adapting UI components

### PR DESCRIPTION
**Pull request by antaresjhw**
### Description

This PR extends the custom components documentation by adding a simple example for how to "override" an existing component to better suit ones needs. It wasn't immediately clear to me how I might adapt the sidebar to use a different project title, since the documentation seems focussed on adding new components and it wasn't obvious to me that adapting existing components is functionally identical.

A big thank you to the Evidence team for getting me pointed in the right direction on doing what I wanted, and I hope that this addition to the docs will help others as well.

### Checklist

- [ ] For UI or styling changes, I have added a screenshot or gif showing before & after
- [ ] I have added a [changeset](https://github.com/evidence-dev/evidence/blob/main/CONTRIBUTING.md#adding-a-changeset)

(I don't believe these checklist items apply since I am updating documentation only - I'll be happy to adapt if needed).

### CODE REVIEW

File: sites/docs/docs/components/custom-components.md
Comment (archiewood) [inline sites/docs/docs/components/custom-components.md:72]: ~~Adapting any existing components to your needs is straightforward as well. You can achieve this by duplicating an existing component into your `/components` directory, modifying that new component file, and importing the new component as needed in your project.~~
Modify an existing component by copying the component's source code into the `/components` directory, modifying it, and importing the new component as needed in your project.

File: sites/docs/docs/components/custom-components.md
Comment (archiewood) [inline sites/docs/docs/components/custom-components.md:71]:
```
## Modifying an existing component: Changing the project name
```

File: sites/docs/docs/components/custom-components.md
Comment (archiewood) [inline sites/docs/docs/components/custom-components.md:93]: you can use ```svelte as a codefence name and github will highlight it correctly

File: sites/docs/docs/components/custom-components.md
Comment (archiewood) [inline sites/docs/docs/components/custom-components.md:106]: ` ```svelte`

File: sites/docs/docs/components/custom-components.md
Comment (archiewood) [inline sites/docs/docs/components/custom-components.md:110]: typo:
...refers to the `/components` ...

**Review by archiewood — CHANGES_REQUESTED** [review]
This is epic, and a really helpful example.

Enough people want to change the project title that I suggest we explicitly call it out in the example, so if people are searching the docs for it, they find it.

We'll probably get to adding this feature pretty soon, so at that point this may be redundant. Until then it's really useful.

A couple of minor wording edits, and I'll merge this!

**Comment by antaresjhw** [comment]
The requested changes all look great and have been resolved.

**Review by ItsMeBrianD — APPROVED** [review]
Looks awesome!

**Comment by archiewood** [comment]
@antaresjhw sweet - if it's not totally clear - you'll actually have to commit the changes  - (clicking "resove conversation" on github doesn't actually do anything)

**Comment by antaresjhw** [comment]
OK, I guess I'm an idiot (or more kindly a GitHub noob). I can't figure out how to actually commit the resolved changes, haha... I'm just using the GitHub web UI. Do I need to do this from my fork? My limited Googling hasn't been much help here, I am afraid.

**Comment by antaresjhw** [comment]
Thanks for the patience - I think I figured it out and was able to get the suggestions in place. I haven't done any cross-fork PR's, so it was a little bit more "manual" than I guess I was expecting.

**Review by archiewood — CHANGES_REQUESTED** [review]
Updates look perfect.

Final thing: There seems to be a dead link in here to `components/pages`. This is causing the docs build to fail. Did you perhaps mean `core-concepts/pages`?

**Comment by antaresjhw** [comment]
Good catch. I made the update, but the build is still failing. It seems like it might be unrelated bo a broken link this time from looking at the details, but I am not completely sure.

**Comment by archiewood** [comment]
That's actually not the docs - those are some other tests on the software itself. They should definitely pass - I just need to kick them off myself (we ensure a maintainer starts these, as a malicious PR could potentially leak credentials etc).

It's annoying that vercel logs this as a failure, rather than eg "waiting for maintainer to approve" - a bit misleading!

**Review by archiewood — APPROVED** [review]

**Pull request merged**
